presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/Partition.java

```diff
@@ -71,6 +74,10 @@ public Partition(
         this.sealedPartition = sealedPartition;
         this.createTime = createTime;
         this.lastDataCommitTime = lastDataCommitTime;
+        if (rowIdPartitionComponent == null) {
```

Comment:
if the concern about null here is due to the serialized object, i believe jackson handles that already (if it's an optional type, serializes empty as null, and reads it back as an empty object).